qiskit/providers/ibmq/job/job_monitor.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""A module for monitoring various qiskit functionality"""
+
+import sys
+import time
+import datetime
+from typing import TextIO
+from .ibmqjob import IBMQJob
+from ..utils.converters import seconds_to_duration
+
+
+def _text_checker(job: IBMQJob,
+                  interval: float,
+                  _interval_set: bool = False,
+                  quiet: bool = False,
+                  output: TextIO = sys.stdout) -> None:
+    """A text-based job status checker.
+
+    Args:
+        job: The job to check.
+        interval: The interval at which to check.
+        _interval_set (bool): Was interval time set by user?
+        quiet (bool): If True, do not print status messages.
+        output (file): The file like object to write status messages to.
+        By default this is sys.stdout.
+
+    """
+    status = job.status()
+    msg = status.value
+    prev_msg = msg
+    msg_len = len(msg)
+    prev_time_str = ''
+    if not quiet:
+        print('\r%s: %s' % ('Job Status', msg), end='', file=output)
+    while status.name not in ['DONE', 'CANCELLED', 'ERROR']:
+        time.sleep(interval)
+        status = job.status()
+        msg = status.value
+
+        if status.name == 'QUEUED':
+            queue_info = job.queue_info()
+
+            time_str = ''
+            if queue_info:
+                est_time = queue_info.estimated_start_time
```

Comment:
Unfortunately `estimated_start_time` can be `None` even if `queue_info` itself isn't, since the API might return only _some_ of the queue information.

Reply:
Fixed by doing a check.